### PR TITLE
Maker: Refactor away from App Lab code

### PR DESCRIPTION
**Pull request by islemaster**
Some general cleanup for issues that came to my attention over the last week.

* Pulls some maker-specific code out of applab/commands.js (which is already too long), moving it into `[cdo]/apps/src/lib/kits/maker/commands.js`.
* Moves some utility modules (javascriptMode.js and rgbcolor.js) to `[cdo]/apps/src/lib/util`
* Adds _slightly_ better test coverage for the moved code.
* Moves the logic for merging maker's `dropletConfig` into applab's `dropletConfig` out of applab.js into applab/dropletConfig.js (closer to how audioApi's dropletConfig gets merged in).

In addition to cleaning up, this gets us a lot closer to being able to share Maker Toolkit with Game Lab by breaking several of the remaining App Lab dependencies.  As of this PR we don't reference `Applab` anywhere within lib/kits/maker.

TODO

* [x] Inject `Applab` into maker/api.js
* [x] Inject `Applab.makerController` into maker/commands.js

### CODE REVIEW

File: apps/src/applab/commands.js [hunk 1]
```diff
@@ -44,25 +45,6 @@ var toBeCached = {};
  */
 var eventSandboxer = new EventSandboxer();
 
-function apiValidateTypeAndRange(opts, funcName, varName, varValue,
```
Comment (islemaster) [inline apps/src/applab/commands.js:47]: This moves to lib/util/javascriptMode.js

File: apps/src/applab/commands.js [hunk 2]
```diff
@@ -1698,51 +1680,6 @@ applabCommands.drawChartFromRecords = function (opts) {
   ).then(onSuccess, onError);
 };
 
-applabCommands.pinMode = function (opts) {
```
Comment (islemaster) [inline apps/src/applab/commands.js:1701]: These move to maker/commands.js.

File: apps/test/unit/lib/kits/maker/dropletConfigTest.js [hunk 3]
```diff
@@ -145,35 +187,4 @@ describe(`timedLoop(ms, callback)`, () => {
     expect(commands).to.haveOwnProperty('timedLoop');
     expect(commands.timedLoop).to.be.a('function');
   });
-
-  it('runs code on an interval', () => {
```
Comment (islemaster) [inline apps/test/unit/lib/kits/maker/dropletConfigTest.js:149]: Moved to commandsTest.js

File: apps/src/lib/util/javascriptMode.js [hunk 4]
```diff
@@ -88,6 +88,25 @@ export function apiValidateType(opts, funcName, varName, varValue, expectedType,
   return !!opts[validatedTypeKey];
 }
 
+export function apiValidateTypeAndRange(opts, funcName, varName, varValue,
+                                        expectedType, minValue, maxValue, opt) {
```
Comment (Bjvanminnen) [inline apps/src/lib/util/javascriptMode.js]: nit: second line should be indented only 4 spaces.

**Pull request merged**